lib/adapters/idb/index.js

```diff
@@ -480,7 +480,16 @@ function init(api, opts, callback) {
 
         var change = opts.processChange(winningDoc, metadata, opts);
         change.seq = metadata.seq;
-        if (filter(change)) {
+
+        var filtered;
+        try {
+          filtered = filter(change);
+        } catch(err) {
+          var msg = 'Filter function threw: ' + err.toString();
+          return opts.complete(errors.error(errors.BAD_REQUEST, msg));
+        }
+
+        if (filtered) {
```

Comment:
So I was thinking of putting this in filterChange but the return type seemed awkward, maybe if it succeeds we coorce the result into a `boolean` and if not return the error object?